base/promotion.jl

```diff
@@ -7,27 +7,23 @@ typejoin(t::ANY) = (@_pure_meta; t)
 typejoin(t::ANY, ts...) = (@_pure_meta; typejoin(t, typejoin(ts...)))
 function typejoin(a::ANY, b::ANY)
     @_pure_meta
-    if isa(a,TypeConstructor); a = a.body; end
-    if isa(b,TypeConstructor); b = b.body; end
     if a <: b
```

Comment:
why is it legal to ask `(a::UnionAll).body <: b` and the reverse? I thought types with free typevars weren't part of the type lattice.